src/rust/engine/client/src/options/id.rs

```diff
@@ -0,0 +1,180 @@
+// Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+use std::fmt;
+use std::fmt::{Display, Formatter};
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum Scope {
+  GLOBAL,
+  Scope(String),
+}
+
+impl Scope {
+  pub fn named(name: &str) -> Scope {
```

Comment:
Constructor naming nit: `with_name`?

Reply:
A scope only has a name and nothing else. I've seen with_X for builders where you set many properties or else "mutate" one property. Is there some precedent for straight up construction like this?

Reply:
There probably is precedent. I'll withdraw my comment.